src/week4/StringAPI.java

```diff
@@ -0,0 +1,142 @@
+        if (input == null || input.isEmpty()) {
+            return OUT_OF_BOUND;
+        }
+        if (k < 0 || k > input.length()) {
+            return OUT_OF_BOUND;
+        }
+        return String.valueOf(input.charAt(k));
+    }
+
+    //Ex2
+    public int getWordCount(String input) {
+        int count = 0;
+        if (input == null || input.isEmpty()) {
+            System.out.println(INVALID);
+            return 0;
+        }
+        if (!(input.matches("(.*) (.*)"))) {
+            System.out.println(" 1 Word ");
+            return 0;
+        }
+        for (int i = 0; i < input.length(); i++) {
+            if (input.charAt(i) == ' ') {
+                count++;
+            }
+        }
+        return count + 1;
+    }
+
+    //Ex3
+    public String checkIfInputIsPalindrome(String input) {
+        StringBuilder st = new StringBuilder(input);
+        if (input == null || input.isEmpty()) {
+            return OUT_OF_BOUND;
+        }
+        if (st.reverse().toString().equals(input)) {
+            return "Palindrome";
+        } else {
+            return "Not Palindrome";
+        }
+    }
+
+    //Ex4
+    public String checkBrackets(String input) {
+        if (input == null || input.isEmpty()) {
+            return INVALID;
+        }
+        int length = input.length();
+        if (length % 2 != 0) {
+            return INVALID;
+        }
+        if (input.matches("[()]+")) {
+            if (input.replace("(", "").length() == length / 2) {
+                return VALID;
+            }
+
+        } else {
+            return INVALID;
+        }
+        return INVALID;
+    }
+
+    //Ex5
+    public boolean isInputContains(String input, CharSequence sequence) {
+        if (input == null || sequence == null || input.isEmpty() || sequence.equals("")) {
+            return false;
+        }
+        if (input.contains(sequence)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    //Ex6
+    public boolean isInputStartWithAnother(String orgString, String input) {
+        if (orgString == null || input == null || orgString.isEmpty() || input.isEmpty()) {
+            return false;
+        }
+        if (orgString.startsWith(input)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    //Ex7
+    public boolean isInputEndWithAnother(String orgString, String input) {
+        if (orgString == null || input == null || orgString.isEmpty() || input.isEmpty()) {
+            return false;
+        }
+        if (orgString.endsWith(input)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    //Ex8
+    public String removeDuplicateChars(String input) {
+        if (input == null || input.isEmpty()) {
+            return INVALID;
+        }
+        char[] arr1 = input.toCharArray();
+        String orgString = "";
+        for (char value : arr1) {
```

Comment:
esi senc ashxatela? mi hat kbacatres sa vonca ashxatum, PR-in el comment anes ok a)

Reply:
ha ashkhatumer
mi hat original string uni vor skzbum empty a
for ov fruma u amen array i index i vra kanga arnum stuguma ed original stringum eghac letternere gtnuma te che, yete chi gtnum -1 a veradarcnum u ed mi hat elementy avelacnuma original stringin, hajort ankam arten hasnuma 2rord elementin u eli mana galis originalum eghac letternere edegh kan te che, yete -1 chveradarcni ed letter e chi avelacni